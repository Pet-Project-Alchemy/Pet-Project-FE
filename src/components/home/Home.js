import React from 'react';
import './Home.scss';
import mp4 from '../assets/dogPhoto/its-clark.mp4';
import webm from '../assets/dogPhoto/its-clark.webm';

const Header = () => {
  return (
    <>
      <div className='background'>
        <header className='header'>
          <div className='bg-video'>
            <video className='bg-video__content' autoPlay muted loop>
              <source src={webm} type='video/webm' alt='dog park video' />
              <source src={mp4} type='video/mp4' alt='dog park video' />
              your browser not supported
            </video>
          </div>
          <div className='header__text-box'>
            <h1 className='heading__primary'>
              <span className='heading-primary__sub--main'>Pet Pals</span>
              <span className='heading-primary__sub'>Let's Go Outside</span>
            </h1>
          </div>
        </header>

        <section className='section__main'>
          <h3 className='section__title'>Learn More</h3>
          <div className='main__container'>
            <p className='main__container--paragraph'>
              We love our pets. Pet Pals builds community and fights loneliness.
              Share the responsibilities of potty breaks or short walks with
              your new Pet Pal. Create your profile, search for dog-lovers in
              your area, use our chat to set up a meeting at the featured local
              dog-friendly spots.
            </p>
          </div>
        </section>
      </div>
    </>
  );
};

export default Header;
