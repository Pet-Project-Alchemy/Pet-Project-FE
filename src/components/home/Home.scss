@import '../abstract/variables';
@import '../abstract/base.scss';
@import '../abstract/animate.scss';

.header {
  height: 95vh;
  .bg-video {
    filter: grayscale(1);
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    z-index: -1;
    clip-path: polygon(0 0, 100% 0, 100% 65vh, 0 100%);

    &__content {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &__text-box {
    position: absolute;
    top: 13%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
  }
}

.heading-primary {
  &__sub--main {
    color: $color-black;
    display: block;
    font-size: 4rem;
    font-weight: 400;
    letter-spacing: 2rem;
    animation: moveInLeft 1s ease-out;
  }
  &__sub {
    display: block;
    color: $color-black;
    font-size: 2rem;
    font-weight: 700;
    letter-spacing: 1.75rem;
    animation: moveInRight 1s ease-in;
  }
}

.section__title {
  color: $color-darkgrey;
  font-size: 4rem;
  padding-top: 2.5rem;
  text-align: center;
}

.main__container {
  display: flex;
  justify-content: center;
  align-items: center;
  background: $color-black;

  &--paragraph {
    display: flex;
    justify-content: center;
    align-items: center;
    color: $color-darkgrey;
    margin: 1rem auto;
    font-size: 2.5rem;
    letter-spacing: 2px;
    height: auto;
    width: 100rem;
    background: $color-black;
    margin: 2rem;
    padding: 2rem;
    border-width: 1rem;
    border-style: solid;
    border-image: linear-gradient(to bottom, $color-primary, rgba(0, 0, 0, 0)) 1
      100%;
  }
}
