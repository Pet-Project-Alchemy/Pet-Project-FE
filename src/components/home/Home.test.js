import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home component', () => {
  it('renders Home', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
