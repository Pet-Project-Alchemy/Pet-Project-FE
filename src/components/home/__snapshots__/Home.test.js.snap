// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders Home 1`] = `
<Fragment>
  <div
    className="background"
  >
    <header
      className="header"
    >
      <div
        className="bg-video"
      >
        <video
          autoPlay={true}
          className="bg-video__content"
          loop={true}
          muted={true}
        >
          <source
            alt="dog park video"
            src="/path/image"
            type="video/webm"
          />
          <source
            alt="dog park video"
            src="/path/image"
            type="video/mp4"
          />
          your browser not supported
        </video>
      </div>
      <div
        className="header__text-box"
      >
        <h1
          className="heading__primary"
        >
          <span
            className="heading-primary__sub--main"
          >
            Pet Pals
          </span>
          <span
            className="heading-primary__sub"
          >
            Let's Go Outside
          </span>
        </h1>
      </div>
    </header>
    <section
      className="section__main"
    >
      <h3
        className="section__title"
      >
        Learn More
      </h3>
      <div
        className="main__container"
      >
        <p
          className="main__container--paragraph"
        >
          We love our pets. Pet Pals builds community and fights loneliness. Share the responsibilities of potty breaks or short walks with your new Pet Pal. Create your profile, search for dog-lovers in your area, use our chat to set up a meeting at the featured local dog-friendly spots.
        </p>
      </div>
    </section>
  </div>
</Fragment>
`;
